src/unix/sock_ctrl_msg.rs

```diff
@@ -443,19 +437,31 @@ mod tests {
     #[test]
     fn buffer_len() {
         assert_eq!(unsafe { CMSG_SPACE(0) as usize }, size_of::<cmsghdr>());
+        #[cfg(any(target_os = "linux", target_os = "android"))]
         assert_eq!(
             unsafe { CMSG_SPACE(size_of::<RawFd>() as c_uint) as usize },
             size_of::<cmsghdr>() + size_of::<c_long>()
         );
+        #[cfg(target_os = "macos")]
+        assert_eq!(
+            unsafe { CMSG_SPACE(size_of::<RawFd>() as c_uint) as usize },
+            size_of::<cmsghdr>() + size_of::<c_uint>()
+        );
         if size_of::<RawFd>() == 4 {
             assert_eq!(
                 unsafe { CMSG_SPACE(2 * size_of::<RawFd>() as c_uint) as usize },
                 size_of::<cmsghdr>() + size_of::<c_long>()
             );
+            #[cfg(any(target_os = "linux", target_os = "android"))]
             assert_eq!(
                 unsafe { CMSG_SPACE(3 * size_of::<RawFd>() as c_uint) as usize },
                 size_of::<cmsghdr>() + size_of::<c_long>() * 2
             );
+            #[cfg(target_os = "macos")]
+            assert_eq!(
+                unsafe { CMSG_SPACE(3 * size_of::<RawFd>() as c_uint) as usize },
+                size_of::<cmsghdr>() + size_of::<c_uint>() * 3
+            );
```

Comment:
Can you add a comment to explain what is the difference and why?

Reply:
okay, so please add the reason in the commit message and maybe also in a comment in the code.

Reply:
> Admittedly, if we're using the libc crate's version of these, this test can probably go away altogether

It has been removed